plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidRecycleRefactoring.java

```diff
@@ -0,0 +1,335 @@
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import static org.autorefactor.refactoring.ASTHelper.*;
+import static org.eclipse.jdt.core.dom.ASTNode.*;
+import org.autorefactor.refactoring.ASTBuilder;
+import org.autorefactor.refactoring.Refactorings;
+
+/*
+ * TODO when the last use of resource is as arg of a method invocation,
+ * it should be assumed that the given method will take care of the release.
+ * TODO Track local variables. E.g., when a TypedArray a is assigned to variable b,
+ * release() should be called only in one variable.
+ * TODO (low priority) check whether resources are being used after release.
+ * TODO add support for FragmentTransaction.beginTransaction(). It can use method
+ * chaining (which means local variable might not be present) and it can be released
+ * by two methods: commit() and commitAllowingStateLoss().
+ */
+
+/** See {@link #getDescription()} method. */
+public class AndroidRecycleRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        return "Many resources, such as TypedArrays, VelocityTrackers, etc., should be "
+                + "recycled (with a recycle()/close() call) after use. " + "Inspired from "
+                + "https://android.googlesource.com/platform/tools/base/+/master/lint/libs/lint-checks/src/main/"
+                + "java/com/android/tools/lint/checks/CleanupDetector.java";
+    }
+
+    @Override
+    public String getName() {
+        return "RecycleRefactoring";
+    }
+
+    private static boolean isMethodIgnoringParameters(MethodInvocation node, String typeQualifiedName,
+            String methodName) {
+        if (node == null) {
+            return false;
+        }
+        final IMethodBinding methodBinding = node.resolveMethodBinding();
+        if (methodBinding == null || !methodName.equals(methodBinding.getName())) {
+            return false;
+        }
+        final ITypeBinding declaringClazz = methodBinding.getDeclaringClass();
+        return instanceOf(declaringClazz, typeQualifiedName);
+    }
+
+    private static boolean isMethodIgnoringParameters(MethodInvocation node, String typeQualifiedName,
+            String[] methodNames) {
+        boolean isSameMethod;
+        for (String methodName : methodNames) {
+            isSameMethod = isMethodIgnoringParameters(node, typeQualifiedName, methodName);
+            if (isSameMethod) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String methodNameToCleanupResource(MethodInvocation node) {
+        if (isMethodIgnoringParameters(
+                node,
+                "android.database.sqlite.SQLiteDatabase",
+                new String[]{"query", "rawQuery", "queryWithFactory", "rawQueryWithFactory"})
```

Comment:
After moving to variable arguments, I think you can get rid of this `new String[] {` ?